kernel/libc/math.c

```diff
@@ -12,25 +13,78 @@ static double reduce_angle(double angle)
 	return angle;
 }
 
+// Taylor series expansion for sine function
 double sin(double x)
 {
 	x = reduce_angle(x);
 
-	// Taylor series expansion for sine function
 	return x - (x * x * x) / 6.0 + (x * x * x * x * x) / 120.0 -
 	       (x * x * x * x * x * x * x) / 5040.0;
 }
 
+// Taylor series expansion for cosine
 double cos(double x)
 {
 	x = reduce_angle(x);
 
-	// Taylor series expansion for cosine function
 	return 1.0 - (x * x) / 2.0 + (x * x * x * x) / 24.0 -
 	       (x * x * x * x * x * x) / 720.0;
 }
 
+// Tangent calculation
 double tan(double x)
 {
 	return sin(x)/cos(x);
 }
+
+// Taylor series expansion for inverse sine function (arcsin)
+double arcsin(double x)
+{
+    return x + ( (exp(x, 3))/6 ) + (( 1 * 3 * exp(x, 5)) / 40.0 ) + ( ( 1 * 3 * 5 * exp(x, 7) ) / 384.0 );
+}
+
+// Factorial
+int factorial(int n)
+{
+    int result = n;
+
+    if (n == 0)
+    {
+	return 1;
+    }
+
+    while (n != 1)
+    {
+	result = result*(n-1);
+	n--;
+    }
+
+    return result;
+}
+
+// Exponentiation
+double exp(double b, double n)
+{
+    double result = 1.0;
+    if (n < 0)
+    {
+	return -1;
+    }
+
+    while (n != 0)
+    {
+	result *= b;
+	n--;
+    }
+    return result;
+}
+
+// Square root
+double sqrt(double x)
+{
+    if (x < 0)
+    {
+	return -1;
+    }
+    return exp(x, 0.5); 
```

Comment:
sqrt should be re-implemented using another approach
[mathematical methods](en.wikipedia.org/wiki/Methods_of_computing_square_roots)
[Newton Raphson Method](https://www.geeksforgeeks.org/newton-raphson-method/)